### PR TITLE
Fix API logging condition

### PR DESCRIPTION
**Pull request by gcusnieux**
**Issue**

https://github.com/gravitee-io/issues/issues/7205

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/7205-fix-generated-logging-condition/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

I also added a small commit to fix form display:

BEFORE:
Capture d’écran 2022-03-03 à 18 55 35

AFTER:
Capture d’écran 2022-03-03 à 18 54 22

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-unaslkupvw.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            LOGGER.info("Skipping {} execution cause it has already been successfully executed", this.getClass().getSimpleName());
+            return false;
+        }
+        if (isStatus(installation, RUNNING)) {
+            LOGGER.warn("Skipping {} execution cause it's already running", this.getClass().getSimpleName());
+            return false;
+        }
+
+        try {
+            LOGGER.info("Starting {} execution with dry-run {}", this.getClass().getSimpleName());
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java]: extra `{}` to remove

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                api.getDefinition(),
+                io.gravitee.definition.model.Api.class
+            );
+            if (
+                apiDefinition.getProxy() != null &&
+                apiDefinition.getProxy().getLogging() != null &&
+                apiDefinition.getProxy().getLogging().getCondition() != null
+            ) {
+                String condition = apiDefinition.getProxy().getLogging().getCondition().trim();
+                if (!condition.isEmpty() && condition.contains("#") && !condition.startsWith("{") && !condition.endsWith("}")) {
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java]: `!condition.isEmpty()` is already covered by `condition.contains` so we can remove it

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            LOGGER.error("{} execution failed", this.getClass().getSimpleName(), e);
+            setExecutionStatus(installation, FAILURE);
+            return false;
+        }
+
+        return true;
+    }
+
+    protected void fixApis() throws Exception {
+        for (Api api : apiRepository.findAll()) {
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java:73]: a little bit afraid of this findAll if customer has big amount of api :(

Comment (ytvnr) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java:73]: tested with 100+ api and it's ok

**Review by ytvnr — APPROVED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java (hunk 2 above)
Comment (gcusnieux) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiLoggingConditionUpgrader.java]: 👍

**Pull request merged**